### PR TITLE
[foxy] Assume root namespace if not provided in ComposableNodeContainer

### LINKED ISSUE #185: [Foxy]: Can't launch a ComposableNodeContainer with the latest Foxy packages (0.10.3)
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 20.04
- Installation type:
  - Foxy debian packages
- Version or commit hash:
  - ros-foxy-launch-ros_0.10.3
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - rclpy

#### Steps to reproduce issue

I have a fairly straightforward launch file:

```
from launch import LaunchDescription
from launch_ros.actions import ComposableNodeContainer
from launch_ros.descriptions import ComposableNode

def generate_launch_description():
    params = {
        'target_frame': 'base_footprint',
        'min_height': 0.10,
        'max_height': 2.10,
        'angle_min': -2.356194490192345,
        'angle_max': 2.356194490192345,
        'angle_increment': 0.005759586531581288,
        'range_min': 0.4,
        'range_max': 200.0,
        'use_inf': False,
    }

    return LaunchDescription([
        ComposableNodeContainer(
            name='velodyne_container',
            namespace='',
            package='rclcpp_components',
            executable='component_container',
            composable_node_descriptions=[
                ComposableNode(package='pointcloud_to_laserscan',
                               plugin='pointcloud_to_laserscan::PointCloudToLaserScanNode',
                               name='pointcloud_to_laserscan',
                               remappings=[('cloud_in', 'velodyne_points')],
                               parameters=[params, {'use_sim_time': True}])
            ],
            output='both',
        ),
    ])
```

#### Expected behavior

Container successfully launches

#### Actual behavior

Throws an exception:

```
$ ros2 launch bug-launch.py 
[INFO] [launch]: All log files can be found below /home/ubuntu/.ros/log/2020-10-19-20-46-35-369504-ros2-focal-1778566
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [component_container-1]: process started with pid [1778568]
Task exception was never retrieved
future: <Task finished name='Task-5' coro=<LaunchService._process_one_event() done, defined at /opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py:271> exception=InvalidServiceNameException("Invalid service name: topic name must not contain characters other than alphanumerics, '_', '~', '{', or '}':\n  '<node_namespace_unspecified>/velodyne_container/_container/load_node'\n   ^")>
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 273, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 293, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/actions/load_composable_nodes.py", line 200, in execute
    self.__rclpy_load_node_client = get_ros_node(context).create_client(
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/node.py", line 1260, in create_client
    self._validate_topic_or_service_name(srv_name, is_service=True)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/node.py", line 1079, in _validate_topic_or_service_name
    validate_topic_name(topic_or_service_name, is_service=is_service)
  File "/opt/ros/foxy/lib/python3.8/site-packages/rclpy/validate_topic_name.py", line 41, in validate_topic_name
    raise InvalidServiceNameException(name, error_msg, invalid_index)
rclpy.exceptions.InvalidServiceNameException: Invalid service name: topic name must not contain characters other than alphanumerics, '_', '~', '{', or '}':
  '<node_namespace_unspecified>/velodyne_container/_container/load_node'
   ^
```

#### Additional information

If I run the above launch file with the latest sources (on master), everything works fine.  If I run the above launch file with the previous version of launch_ros (0.10.2), everything works fine.  There is only one patch between 0.10.2 and 0.10.3, and it comes from this PR: https://github.com/ros2/launch_ros/pull/179 .  So it looks like something in that backport is either missing or went wrong.

@ivanpauno @hidmic FYI.

### PR DESCRIPTION
**Pull request by jacobperron**
Fixes a regression introduced in #179

Fixes #185.

We apply the same logic in LifecycleNode to maintain backwards compatibility in Foxy.

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
This certainly fixes my example in #185, so I'll approve.  I'd like to get another approval from @hidmic or @ivanpauno before merging, though.  (and run CI, of course)

**Review by ivanpauno — APPROVED** [review]
Sounds reasonable to me!

Maybe we should ignore the suggestion I made [here](https://github.com/ros2/launch_ros/pull/179#discussion_r492014290), so this is also the case in `Node` constructor.
But this fix the issue without introducing the risk of another backward compatibility problem, so LGTM!

**Comment by jacobperron** [comment]
> Maybe we should ignore the suggestion I made here, so this is also the case in Node constructor

I can look into it, with more testing.

**Comment by jacobperron** [comment]
Testing `launch_ros` and `test_launch_ros`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12802/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7761/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10516/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12749/)

**Pull request merged**
